sdk/servicebus/Azure.Messaging.ServiceBus/src/Receiver/ServiceBusReceiverClient.cs

```diff
@@ -459,18 +470,10 @@ public virtual async Task<ServiceBusMessage> PeekAsync(CancellationToken cancell
             if (IsSessionReceiver)
             {
                 // if this is a session receiver, the receive link must be open in order to peek messages
-                await RetryPolicy.RunOperation(
-                    async (timeout) => await Consumer.ReceiveLink.GetOrCreateAsync(timeout).ConfigureAwait(false),
-                    EntityName,
-                    Consumer.ConnectionScope,
-                    cancellationToken).ConfigureAwait(false);
+                await Consumer.GetOrCreateLinkAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Don't we still need retries here?

Reply:
In `AmqpConsumer` class `GetOrCreateLinkAsync` method was already there and it has retry logic.

Reply:
Ah, nevermind!